### PR TITLE
Update validation schemas

### LINKED ISSUE #595: Unifying Sign In and Sign Up Password Validation Schemes
### Overview
Currently Sign In and Sign Up have different special character validation schemes, preventing password unification. 

### Action Items
- unify the Sign In and Sign Up password validation scheme for special characters - currently the Sign In password scheme does not take special characters like `*` and `;`

### PR DESCRIPTION
**Pull request by mira-kine**
Closes #595 

### What changes did you make?
  - Updated regex in `SignInForm` with special characters regex in `PasswordValidationSchema`

### Testing done for these changes
  - Tried signing up with special characters not explicit in SignIn but passes in SignUp, then tried signing back in. SignIn button was disabled until I used the same regex code for both.

### CODE REVIEW

**Review by paulespinosa — CHANGES_REQUESTED** [review]
@mira-kine Good start! Why isn't utils password validation schema being used here? And please add tests.

Here's some ideas for test cases:
```
test123!
Testing!
Test123!
Test1234
12345678
7væVPj¼±mó5÷ÙÞW
Xin chào 1 bạn!
```

File: app/src/components/authentication/SignInForm.tsx [hunk 1]
```diff
@@ -33,10 +33,7 @@ const validationSchema = object({
       /^(?=.*[A-Z])/,
       'password must contain at least one uppercase character',
     )
-    .matches(
-      /^(?=.*[!@#%&])/,
-      'password must contain at least one special character',
-    ),
+    .matches(/\W|_/g, 'password must contain at least one special character'),
```
[the remark this answers is not in the document]
Comment (mira-kine) [reply app/src/components/authentication/SignInForm.tsx:36]: Good call! I'll go ahead and change that.

**Comment by mira-kine** [comment]
@paulespinosa the utils validation schemas is not shared across the components because we are validating differently here. In SignUp, we are using Formik + Yup to check that a user inputs a valid password before it is entered into the db, but for SignIn, we are only validating that there is a password at all. We had made this change after I made my PR, so I forgot to update it. Good catch - the PR should be updated with these changes now. I also updated the PR with the regex changes. Will be pushing up tests on SignUpForm special characters next.

**Comment by paulespinosa** [comment]
@mira-kine Got it, makes sense. Thank you! The changes look good so far.

**Review by Joshua-Douglas — CHANGES_REQUESTED** [review]
Hey @mira-kine, 

Thanks for the changes. The code changes look great, but the test cases are failing and I have some comments on the docs. 

I made some suggestions that should help fix the test cases but I have a few more observations:

1. Running the test cases gives this warning:

```ts
Warning: An update to PasswordValidation inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
  /* fire events that update state */
});
/* assert on the output */
```

I was able to dismiss it by wrapping the render and fireEvent calls in the `act` block. I'm not sure if there is a more elegant solution. 

2. `SignInForm.should display error text` test case is failing because the login form no longer contains the `password must contain` message. We removed that message in this PR, so you can remove that check. 
3. I couldn't get the `7væVPj¼±mó5÷ÙÞW` test case to pass, even though this validation passes in the browser. I suspect there is an issue with the testing code, but I couldn't find it. Maybe @erikguntner might see the issue if you can't sort it out.

File: app/src/components/__tests__/SignUpForm.test.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    const {emailInput, passwordInput} = prepare();
+    expect(emailInput).toBeInTheDocument();
+    expect(passwordInput).toBeInTheDocument();
+  });
+
+  // when correct special characters are added, it should pass validation form
+  it('should pass password validation form with correct special characters', () => {
+    const {passwordInput, passwordValidation} = prepare();
+    fireEvent.change(passwordInput, {target: {value: 'test123!'}});
+    expect(passwordValidation).not.toBeInTheDocument();
```
[the remark this answers is not in the document]
Comment (mira-kine) [reply app/src/components/__tests__/SignUpForm.test.tsx:48]: Thank you so much for your suggestions! Now that I have come back with fresh eyes, I noticed that my whole test approach was wrong. I will request reviews once more.

**Review by tylerthome — APPROVED** [review]

**Review by paulespinosa — APPROVED** [review]

**Pull request merged**